components/contribution-flow/index.js

```diff
@@ -74,75 +75,41 @@ const STEP_LABELS = defineMessages({
   },
 });
 
-class ContributionFlow extends React.Component {
-  static propTypes = {
-    collective: PropTypes.shape({
-      slug: PropTypes.string.isRequired,
-      currency: PropTypes.string.isRequired,
-      platformContributionAvailable: PropTypes.bool,
-      parent: PropTypes.shape({
-        slug: PropTypes.string,
-      }),
-    }).isRequired,
-    host: PropTypes.object.isRequired,
-    tier: PropTypes.object,
-    intl: PropTypes.object,
-    createOrder: PropTypes.func.isRequired,
-    confirmOrder: PropTypes.func.isRequired,
-    fixedInterval: PropTypes.string,
-    fixedAmount: PropTypes.number,
-    platformContribution: PropTypes.number,
-    skipStepDetails: PropTypes.bool,
-    loadingLoggedInUser: PropTypes.bool,
-    hasNewPaypal: PropTypes.bool,
-    isEmbed: PropTypes.bool,
-    step: PropTypes.string,
-    redirect: PropTypes.string,
-    verb: PropTypes.string,
-    contributeAs: PropTypes.string,
-    defaultEmail: PropTypes.string,
-    defaultName: PropTypes.string,
-    /** @ignore from withUser */
-    refetchLoggedInUser: PropTypes.func,
-    /** @ignore from withUser */
-    LoggedInUser: PropTypes.object,
-    createCollective: PropTypes.func.isRequired, // from mutation
-    router: PropTypes.object,
-  };
-
-  constructor(props) {
-    super(props);
-    this.mainContainerRef = React.createRef();
-    this.formRef = React.createRef();
-    this.state = {
-      error: null,
-      stripe: null,
-      braintree: null,
-      isSubmitted: false,
-      isSubmitting: false,
-      stepProfile: null,
-      stepPayment: null,
-      stepSummary: null,
-      showSignIn: false,
-      stepDetails: {
-        quantity: 1,
-        interval: props.fixedInterval || props.tier?.interval,
-        amount: props.fixedAmount || getDefaultTierAmount(props.tier),
-        platformContribution: props.platformContribution,
-      },
-    };
-  }
-
-  componentDidUpdate(oldProps) {
-    if (oldProps.LoggedInUser && !this.props.LoggedInUser) {
-      this.setState({ stepProfile: null, stepSummary: null, stepPayment: null });
-      this.pushStepRoute(STEPS.PROFILE);
+function ContributionFlow(props) {
```

Comment:
Do we really need to convert that to a functional component? It creates a massive diff, and potential side-effects are difficult to predict. For example, I can already tell by looking at the code that memorization is broken for all functions using `memoizeOne`. 

Unless there's a good argument for moving to a functional component, we should revert the changes on this file. Class components are accepted in our codebase, we shouldn't try to migrate everything unless it brings clear value.

Reply:
I agree. Addressed in https://github.com/opencollective/opencollective-frontend/pull/6193 😄